src/session/content/message_row/label.rs

```diff
@@ -0,0 +1,244 @@
+use gtk::prelude::*;
+use gtk::subclass::prelude::*;
+use gtk::{glib, pango, CompositeTemplate};
+
+use crate::session::content::message_row::MessageIndicators;
+
+const OBJECT_REPLACEMENT_CHARACTER: char = '\u{FFFC}';
+const INDICATORS_SPACING: i32 = 6;
+
+mod imp {
+    use super::*;
+    use once_cell::sync::Lazy;
+    use once_cell::unsync::OnceCell;
+    use std::cell::RefCell;
+
+    #[derive(Debug, Default, CompositeTemplate)]
+    #[template(string = r#"
+    <interface>
+      <template class="MessageLabel" parent="GtkWidget">
+        <child>
+          <object class="GtkLabel" id="label">
+            <property name="use-markup">True</property>
+            <property name="wrap">True</property>
+            <property name="wrap-mode">word-char</property>
+            <property name="xalign">0</property>
+            <property name="yalign">0</property>
+          </object>
+        </child>
+      </template>
+    </interface>
+    "#)]
+    pub(crate) struct MessageLabel {
+        pub(super) text: RefCell<String>,
+        pub(super) indicators: OnceCell<MessageIndicators>,
+        pub(super) indicators_size: RefCell<Option<(i32, i32)>>,
+        #[template_child]
+        pub(super) label: TemplateChild<gtk::Label>,
+    }
+
+    #[glib::object_subclass]
+    impl ObjectSubclass for MessageLabel {
+        const NAME: &'static str = "MessageLabel";
+        type Type = super::MessageLabel;
+        type ParentType = gtk::Widget;
+
+        fn class_init(klass: &mut Self::Class) {
+            Self::bind_template(klass);
+            klass.set_css_name("messagelabel");
+        }
+
+        fn instance_init(obj: &glib::subclass::InitializingObject<Self>) {
+            obj.init_template();
+        }
+    }
+
+    impl ObjectImpl for MessageLabel {
+        fn properties() -> &'static [glib::ParamSpec] {
+            static PROPERTIES: Lazy<Vec<glib::ParamSpec>> = Lazy::new(|| {
+                vec![
+                    glib::ParamSpecString::new(
+                        "label",
+                        "Label",
+                        "The label of the widget",
+                        None,
+                        glib::ParamFlags::READWRITE | glib::ParamFlags::EXPLICIT_NOTIFY,
```

Comment:
It crashes because by default the label in most places is None (e.g. in the `MessageText`). Anyway, we don't need that flag because we don't really care if the label is not setted in the constructing phase.